### DISCUSSION TITLE
Getting component type JSON dynamically

### DISCUSSION THREAD

**Discussion by gustavohleal** [GrapesJS/grapesjs#5146] [Q&A]
Hi!

I'm working on a converter from an old HTML generated by an old editor from my company. We already have an converter that gets that old html and creates a new html based on grapes old way to load a project. Now i'm want to make this converter generate a projectData like JSON to better load and avoid some errors that was occurring. 

So, my question here is:
Is there a way to get the JSON form generated by the `editor.getProjectData()` function from the component type that exists in the `DomComponents` without having the component rendered in the canvas?

Something like:
```javascript
    let imageType = editor.DomComponents.getType('image')
    // returns the image type model
   let imageJson = imageType.getComponentJson()
 /* returns some like this structure
    {
        "type": "image",
        ...others grapes attributes
        "attributes": {
            "src": "defaultImageURL",
            "alt": "image",
            ... others default attributes
        }
    }
  */
```

If there is not a way to do this by default, where could I find in the source code the generation of the projectData to get some ideas of how to do this? 

Thanks

**Comment by artf** [comment]
You can't have the JSON of the component if the component is not created. You can always create one, get its JSON and then remove it from the document

**Reply by gustavohleal** [reply]
Ok. And is there a way to get the component JSON from the document without using `editor.getProjectData()`?

**Reply by artf** [reply]
You can do this: `JSON.parse(JSON.stringify(component))`
